lib/yuriita/exclusive_collection.rb

```diff
@@ -25,24 +25,17 @@ def view_options
     def view_option(option)
       ViewOption.new(
         name: option.name,
-        selected: selected?(option),
+        selected: selector.selected?(option),
         params: params(option),
       )
     end
 
-    def selected?(option)
-      # We have to do this because of how we're creating the options.
-      # In the definition we are creating duplicate options when we call the
-      # method multiple times.
-      active_option.name == option.name
```

Comment:
This is the TODO item mentioned in the description that was fixed. The new implementation is in `ExclusiveSelect#selected?`. It now uses object comparison on the `Option` objects. `Option` now defines `==` to match based on class and Name.